### PR TITLE
script to show and manage surfshark VPN connection

### PR DESCRIPTION
**Pull request by jordicl**
## Description

These 3 scripts are used to manage your Surfshark VPN connection. It can show the current status inline, connect and disconnect the VPN.

## Type of change

- [x] New script command

## Screenshot

Screenshot 2021-03-04 at 23 00 31

## Dependencies / Requirements

This script depends on the `vpnutil` command line utility. This utility can be downloaded here: https://github.com/Timac/VPNStatus/releases/tag/1.0. Please make sure it's added to your `$PATH`.

## Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: commands/apps/surfshark/surfshark-vpn-start.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Surfshark VPN connect TEST
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark-vpn-start.sh:5]:
```suggestion
# @raycast.title VPN Connect
```

File: commands/apps/surfshark/surfshark-vpn-start.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# @raycast.mode compact
+# @raycast.packageName VPN
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark-vpn-start.sh:7]:
```suggestion
# @raycast.packageName Surfshark
```

File: commands/apps/surfshark/surfshark-vpn-status.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Surfshark VPN status TEST
+# @raycast.mode inline
+# @raycast.refreshTime 10s
+# @raycast.packageName VPN
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark-vpn-status.sh:8]:
```suggestion
# @raycast.packageName Surfshark
```

File: commands/apps/surfshark/surfshark-vpn-status.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Surfshark VPN status TEST
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark-vpn-status.sh:5]:
```suggestion
# @raycast.title Surfshark VPN Status
```

File: commands/apps/surfshark/surfshark-vpn-stop.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Surfshark VPN disconnect TEST
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark-vpn-stop.sh:5]:
```suggestion
# @raycast.title VPN Disconnect
```

File: commands/apps/surfshark/surfshark-vpn-stop.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+# @raycast.mode compact
+# @raycast.packageName VPN
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark-vpn-stop.sh:7]:
```suggestion
# @raycast.packageName Surfshark
```

**Review by unnamedd — CHANGES_REQUESTED** [review]
Hi @jordicl,

yesterday I reviewed a similar [pull request](https://github.com/raycast/script-commands/pull/277) and I made some suggestions there which fit here as well.

Please, check this [comment](https://github.com/raycast/script-commands/pull/277#pullrequestreview-605045848) I made.

Basically, I would like to you to create a new file a config file and move from each file you have, this piece of code to the new config file: 

let's for example, call it `surfshark.config.sh`:
```shell
# You can find the VPN name in the Network preference pane 
export SURFSHARK_VPN_NAME=""

if ! command -v vpnutil &> /dev/null; then
  echo "vpnutil command is required. Download here: https://github.com/Timac/VPNStatus/releases/tag/1.0";
  exit 1;
fi

if [ -z "$SURFSHARK_VPN_NAME" ]; then
  echo "\$SURFSHARK_VPN_NAME is empty. Please, set VPN name in \"surfshark.config.sh\".";
  exit 1;
 fi
```

and in each file, where we had the piece of code above, add this in replacement:
```shell
source surfshark.config.sh

VPN=$SURFSHARK_VPN_NAME
```

In my point of view, this will reduce the amount of duplicated code and concentrate the configuration of the Script Command on the same place and if, something goes wrong, the user will change only one file that's all.

Thanks for your contribution!

File: commands/apps/surfshark/surfshark.config.sh [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# You can find the VPN name in the Network preference pane 
+export SURFSHARK_VPN_NAME="Surfshark. IKEv2"
```
Comment (unnamedd) [inline commands/apps/surfshark/surfshark.config.sh:4]: My suggestion for you is to let this string empty to force the user to configure its own VPN name, or does that the default name for the Surfshark VPN networks?

Comment (jordicl) [reply commands/apps/surfshark/surfshark.config.sh:4]: I left it in as this is the default string generated by the Surfshark app.

Comment (unnamedd) [reply commands/apps/surfshark/surfshark.config.sh:4]: I see. Then, looks that is everything okay! 🤘

**Review by unnamedd — APPROVED** [review]
Looks very good to me, with no duplicated code! It's approved! ✅ 
@thomaspaulmann @PitNikola

**Comment by grzegorzkrukowski** [comment]
It looks good - @jordicl let's just update headers for all scripts, to include dependency information (`vpnutil`) according to our guidelines:
https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md#scripts-that-require-installation-of-runtimes-and-dependencies

and we will merge it.

**Review by grzegorzkrukowski — CHANGES_REQUESTED** [review]
Add dependency informations in headers for all scripts

**Comment by grzegorzkrukowski** [comment]
All good - thanks for quite updates

**Review by grzegorzkrukowski — APPROVED** [review]

**Pull request merged**